src/components/RelatedItems.js

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from "react";
+import { API, graphqlOperation } from "aws-amplify";
+import * as queries from "../graphql/queries";
+import { arkLinkFormatted } from "../lib/MetadataRenderer";
+import { NavLink } from "react-router-dom";
+import Slider from "react-slick";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
+
+import "../css/RelatedItems.css";
+
+class RelatedItems extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: null
+    };
+  }
+
+  async getItems(collection_id, item_limit) {
+    const items = await API.graphql(
+      graphqlOperation(queries.searchArchives, {
+        filter: {
+          heirarchy_path: { eq: collection_id }
+        },
+        sort: {
+          field: "identifier",
+          direction: "asc"
+        },
+        limit: item_limit
+      })
+    );
+    return items.data.searchArchives.items;
+  }
+
+  removeDuplicates(item_list) {
+    let uniqueItems = item_list.filter(
+      (item, index, array) => array.findIndex(i => i.id === item.id) === index
+    );
+    return uniqueItems;
+  }
+
+  removeItem(item_list) {
+    let arr = item_list.filter(
+      value => value.identifier !== this.props.collection.identifier
+    );
+    return arr;
+  }
```

Comment:
Can you merge these two functions into one?

Reply:
Yes!